src/map/atcommand.c

```diff
@@ -2798,73 +2802,89 @@ ACMD(hatch)
 	return true;
 }
 
-/*==========================================
+/**
+ * Sets pet's intimacy value.
  *
- *------------------------------------------*/
+ * @code{.herc}
+ *	@petfriendly <0-1000>
+ * @endcode
+ *
+ */
 ACMD(petfriendly)
 {
-	int friendly;
-	struct pet_data *pd;
-
-	if (!*message || (friendly = atoi(message)) < 0) {
-		clif->message(fd, msg_fd(fd,1016)); // Please enter a valid value (usage: @petfriendly <0-1000>).
+	if (*message == '\0' || (atoi(message) == 0 && !isdigit(*message))) {
+		clif->message(fd, msg_fd(fd, 1016)); /// Please enter a valid value (usage: @petfriendly <0-1000>).
 		return false;
 	}
 
-	pd = sd->pd;
-	if (!pd) {
-		clif->message(fd, msg_fd(fd,184)); // Sorry, but you have no pet.
+	int friendly = atoi(message);
+
+	if (friendly < PET_INTIMACY_NONE || friendly > PET_INTIMACY_MAX) {
```

Comment:
instead of checking for min/max value, how about just cap the value and let it proceed?
```
friendly = cap_value(friendly, PET_INTIMACY_NONE, PET_INTIMACY_MAX);
```

Reply:
I think throwing an error message here, to notifiy the user, is the better way,

Reply:
Imo, I find this this way are "more friendly". 
For example: current commands like `@zeny 123` give 100 Zeny or  `@stats 100` would give +100 stat, and `@stats 129837` would give me a max stats (+32767), and wont block me and throw a error message like `Please type number between 1~32767`. 
However it does have some cons that give negative result due to overflowed numbers like `@zeny 102938120983019283` may result in negative amount.

Reply:
@Emistry: I agree with you in most cases, but an example: The user's intention was to set intimacy to 900 but accidentally wrote 9000. When capping the value, the intimacy would be set to max, even though this was not what user wanted to do. So when whrowing an error message, there's at least a chance to correct this.
BTW; You don't have to instantly resolve a conservation just because I have another opinion. You're welcome to throw in your suggestion and discuss it. I'm always open for suggestions and advices. :-)

Reply:
> The user's intention was to set intimacy to 900 but accidentally wrote 9000. When capping the value, the intimacy would be set to max, even though this was not what user wanted to do. So when throwing an error message, there's at least a chance to correct this.

ah i see. 
about the resolve conversation, its ok, I just randomly throwing question/opinion sometime. no worry :D everything good.